cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -58,6 +60,59 @@ def if_scvmm_refresh_provider(provider):
         provider.refresh_provider_relationships()
 
 
+def check_power_options_when_on(soft_assert, vm, from_details):
```

Comment:
looking at the code it seems there is a lot of structural reptition

@psav should we take a look at simplifying those structures in future?